llvm/lib/CodeGen/SelectionDAG/LegalizeIntegerTypes.cpp

```diff
@@ -2760,9 +2760,19 @@ void DAGTypeLegalizer::ExpandIntegerResult(SDNode *N, unsigned ResNo) {
   LLVM_DEBUG(dbgs() << "Expand integer result: "; N->dump(&DAG));
   SDValue Lo, Hi;
   Lo = Hi = SDValue();
+  bool NeedCustomLower = true;
+
+  // Skip the following function 'CustomLowerNode' when the operand had done
+  // `SoftenFloatResult`.
+  if (N->getOpcode() == ISD::BITCAST &&
```

Comment:
Shouldn't be special casing a specific opcode here, and shouldn't need to check the type of the actual softened result. It is also OK to request custom lowering on illegal types